clients/src/test/java/org/apache/kafka/common/config/AbstractConfigTest.java

```diff
@@ -434,6 +434,22 @@ public void testAutoConfigResolutionWithDuplicateConfigProvider() {
         assertEquals(config.originals().get("sasl.kerberos.key"), "${file:/usr/kerberos:key}");
     }
 
+    @Test
+    public void testConfigProviderConfigurationWithConfigParams() {
+        // Test Case: Valid Test Case With Multiple ConfigProviders as a separate variable
+        Properties providers = new Properties();
+        providers.put("config.providers", "vault");
+        providers.put("config.providers.vault.class", "org.apache.kafka.common.config.provider.MockVaultConfigProvider");
```

Comment:
Could use `MockVaultConfigProvider.class.getName()` instead of hard-coding the class name so that changes are reflected automatically.